--- lib/python2.7/site-packages/tw2/dynforms/widgets.py	2014-07-24 21:14:27.906655298 +0000
+++ lib/python2.7/site-packages/tw2/dynforms/widgets.py	2014-07-24 21:28:24.566339799 +0000
@@ -123,7 +123,7 @@
                     for v in c.value:
                         show.update(c.mapping.get(v, []))
                 else:
-                    show.update(c.mapping.get(c.value, []))
+                    show.update(c.mapping.get(str(c.value), []))
             if c.id in self.hiding_ctrls and c.id not in show:
                 c.safe_modify('container_attrs')
                 c.container_attrs['style'] = 'display:none;' + c.container_attrs.get('style', '')
@@ -148,7 +148,13 @@
                     else:
                         data[c.id] = c._validate(value.get(c.id), data)
                         if isinstance(c, HidingComponentMixin):
-                            show.update(c.mapping.get(data[c.id], []))
+			    #Support Multiselect List widgets
+			    #Show widgets based on pre-validated value
+                            if isinstance(data[c.id],list):
+                                for item in data[c.id]:
+                                    show.update(c.mapping.get(item,[]))
+                            else:
+                                show.update(c.mapping.get(value.get(c.id), []))
                 except twc.ValidationError:
                     data[c.id] = twc.Invalid
                     any_errors = True
